src/parameters.jl

```diff
@@ -14,13 +14,102 @@ An abstract type that represents a set of multiple parameters.
 
 abstract type AbstractParameterSet end
 
+"""Returns the number of parameters in a parameter set."""
+function length(::T) where {T <: AbstractParameterSet}
+  return length(fieldnames(T))
+end
+
+"""Returns the name of the parameters in a parameter set."""
+function names(parameter_set::T) where {T <: AbstractParameterSet}
+  n = Vector{String}(undef, length(parameter_set))
+  return names!(parameter_set, n)
+end
+
+"""Returns the name of the parameters in a parameter set in place."""
+function names!(parameter_set::T, vals::Vector{String}) where {T <: AbstractParameterSet}
+  length(parameter_set) == length(vals) || error("Error: 'vals' should have length $(length(parameter_set)), but has length $(length(vals)).")
+  for (i, param_name) in enumerate(fieldnames(T))
+    p = getfield(parameter_set, param_name)
+    vals[i] = name(p)
+  end
+  return vals
+end
+
+"""Set names of parameters to their fieldnames."""
+function set_names!(parameter_set::T) where {T <: AbstractParameterSet}
+  for param_name in fieldnames(T)
+    p = getfield(parameter_set, param_name)
+    p.name = string(param_name)
+  end
+end
+
+"""Returns current values of each parameter in a parameter set."""
+function values(parameter_set::T) where {T <: AbstractParameterSet}
+  vals = Vector{Any}(undef, length(parameter_set))
+  return values!(parameter_set, vals)
+end
+
+"""Returns current values of each parameter in a parameter set in place."""
+function values!(parameter_set::T, vals::AbstractVector) where {T <: AbstractParameterSet}
+  length(parameter_set) == length(vals) || error("Error: 'vals' should have length $(length(parameter_set)), but has length $(length(vals)).")
+  for (i, param_name) in enumerate(fieldnames(T))
+    p = getfield(parameter_set, param_name)
+    vals[i] = value(p)
+  end
+  return vals
+ end
+
+"""Updates the values of a parameter set by the values given in a vector of values."""
+function update!(parameter_set::T, new_values::AbstractVector) where {T <: AbstractParameterSet}
+  length(parameter_set) == length(new_values) || error("Error: 'new_values' should have length $(length(parameter_set)), but has length $(length(new_values)).")
+  for (i, param_name) in enumerate(fieldnames(T))
+    p = getfield(parameter_set, param_name)
+    converted_value = convert(p, new_values[i])
+    set_value!(p, converted_value)
+  end
+end
```

Comment:
@tmigot This is the function I decided to go for to modify in place a parameter_set with a given vector of values. Instead of building a new parameter set each time, I'll modify one in place. `Parameter{T, D}` is a mutable struct so it shouldn't be a problem